src/formatDate.js

```diff
@@ -50,7 +50,36 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const mainDate = date.split(fromFormat[fromFormat.length - 1]);
+  const objDate = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    objDate[fromFormat[i]] = mainDate[i];
+
+    if (fromFormat[i] === 'YY' && mainDate[i] >= 30) {
+      objDate.YYYY = '19' + mainDate[i];
+    }
+
+    if (fromFormat[i] === 'YY' && mainDate[i] < 30) {
+      objDate.YYYY = 20 + mainDate[i];
+    }
+
+    if (fromFormat[i] === 'YYYY') {
+      objDate.YY = mainDate[i].slice(2);
+    }
+  }
+
+  for (let i = 0; i < toFormat.length; i++) {
+    for (const key in objDate) {
+      if (key === toFormat[i]) {
+        toFormat[i] = objDate[key];
+      }
+    }
+  }
+
+  const result = toFormat.slice(0, -1).join(toFormat[toFormat.length - 1]);
```

Comment:
Use destructuring to get the separator from 'toFormat'. For example: const [separator] = toFormat.slice(-1); Then, replace join(toFormat[toFormat.length - 1]) with join(separator).